### ISSUE TITLE
CPU Profiling: Unexpected end of JSON input

### ISSUE THREAD

**Issue by UncleFirefox** [microsoft/vscode#169001]
Type: <b>Bug</b>

1) Spin up a js process you can debug
2) Use console.profile()
3) Try open the .cpuprofile generated

Expected: I can see a window with the information I see here: https://code.visualstudio.com/docs/nodejs/profiling#_analyzing-a-profile

Actual: I get `Unexpected end of JSON input`

NOTES: 
I can open the file ok using chrome
Using the Record button seems to work fine
Seems to be related to this: https://github.com/microsoft/vscode/issues/148956

VS Code version: Code 1.74.0 (5235c6bb189b60b01b1f49062f4ffa42384f8c91, 2022-12-05T16:47:18.476Z)
OS version: Linux x64 5.15.0-56-generic snap
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor (24 x 2200)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off|
|Load (avg)|0, 0, 0|
|Memory (System)|31.27GB (22.26GB free)|
|Process Argv|--no-sandbox --force-user-env --unity-launch --crash-reporter-id f24231db-955a-4bb8-a229-d037dad599a4|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (38)</summary>

Extension|Author (truncated)|Version
---|---|---
cppincludeguard|aki|1.6.0
file-properties-viewer|bru|1.0.0
vscode-eslint|dba|2.2.6
gitlens|eam|13.1.1
prettier-vscode|esb|9.10.3
go|gol|0.36.0
cloudcode|goo|1.21.0
vscode-test-explorer|hbe|2.21.1
tabsanity|jed|0.0.11
cmake-language-support-vscode|jos|0.0.7
vscode-codemetrics|kis|1.24.0
vscode-catch2-test-adapter|mat|3.0.0
vscode-language-babel|mgm|0.0.36
git-graph|mhu|1.30.0
vscode-docker|ms-|1.23.2
vscode-dotnet-runtime|ms-|1.6.0
playwright|ms-|1.0.1
isort|ms-|2022.8.0
python|ms-|2022.20.1
vscode-pylance|ms-|2022.12.20
jupyter|ms-|2022.11.1003412109
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.12
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.266.1
cmake-tools|ms-|1.12.27
cpptools|ms-|1.13.7
test-adapter-converter|ms-|0.1.6
vscode-js-profile-flame|ms-|1.0.5
vscode-yaml|red|1.10.1
pdf|tom|1.2.1
cmake|twx|0.0.17
intellicode-api-usage-examples|Vis|0.2.6
vscodeintellicode|Vis|1.2.29
vscode-icons|vsc|12.0.1
volar|Vue|1.0.13
vscode-proto3|zxh|0.5.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627cf:30244335
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vscaat:30438848
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyonecf:30548226
pyindex848cf:30577861
nodejswelcome1:30587005
3biah626:30602489
gswce1:30612156
3d0df643:30613357
dbltrim-noruby:30604474
f6dab269:30613381
fim-prod:30623723

```

</details>

**Comment by connor4312** [comment]
Can you share the resulting `.cpuprofile` file?

**Comment by tmkn** [comment]
I got the same error but I think it's due to trying to open the profile via the diff editor. If I open it via the explorer it displays fine.

When I click on it via the Source Control Panel:
![image](https://user-images.githubusercontent.com/2671613/208239786-b5362585-677f-4c56-a8ba-46f8e88f7a99.png)

Clicking "Try Again":
![image](https://user-images.githubusercontent.com/2671613/208239817-c5e99b6b-69cf-4d53-b2c6-e000b402f063.png)
Clicking Ok closes the view again to display the view from the 1st screenshot.

Selecting the file via the Explorer works fine:
![image](https://user-images.githubusercontent.com/2671613/208239873-e4496f06-f6e8-4d9b-b560-d01f160c062c.png)

The only reason why I tried to open it via the Source Control Panel was that I didn't know where it would put the profile and I didn't want to search it in the Explorer. Instead I went to the Source Control Panel, since it will prominently show up as it's untracked.
